contracts/VotableStakingRewards.sol

```diff
@@ -244,6 +229,112 @@ contract VotableStakingRewards is
     emit RewardsDurationUpdated(rewardsDuration);
   }
 
+  /* ========== VOTER FUNCTIONS ========== */
+    
+    /*development:
+      1. set up locking feature
+      2. make sure voter can only allocate up to the amount of ube they have
+      3. calculate the weights at the end of the locking period
+    */
+
+
+uint256 lockDuration = 6 days;
+uint256 lockTime;
+mapping(address => uint256) userLocked;
+mapping(address => mapping(uint256 => uint256)) userWeights;
+mapping(uint256 => uint256) poolWeights;
+
+function getPoolWeight(uint256 poolId) public view returns(uint256) {
+  return poolWeights[poolId];
+}
+
+function allocate_pool_weight(uint256 poolId, uint256 amount) external{
+  require(this.isLocked());
+  require(amount > 0, "Cannot allocate 0");
+  Voter v = voters[msg.sender];
+  require(rewardsToken.balanceOf(address(v)) - userLocked[address(v)] >= amount, "Cannot allocate more than you have");
```

Comment:
very, very dangerous. If the user's balance is less than the amount locked, the subtraction will underflow and this require will evaluate to `true`